pkg/applicationserver/io/packages/loradms/v1/package.go

```diff
@@ -51,6 +51,11 @@ func (p *DeviceManagementPackage) HandleUp(ctx context.Context, assoc *ttnpb.App
 	ctx = log.NewContextWithField(ctx, "namespace", "applicationserver/io/packages/loradms/v1")
 	logger := log.FromContext(ctx)
 
+	if up.DevEUI == nil || up.DevEUI.IsZero() {
+		logger.Warn("Package configured for end device with no device EUI")
```

Comment:
This is not an infrastructural warning; it's debug or at most info level.

The nice way to communicate this to users is via an event that a package failed to process a message.